pandas/core/frame.py

```diff
@@ -654,6 +654,8 @@ def __init__(
             data = data._mgr
 
         if isinstance(data, (BlockManager, ArrayManager)):
+            if using_copy_on_write():
+                data = data.copy(deep=False)
```

Comment:
The previous PR only did this for DataFrame, not for BlockManager. 

We have many places where we have the pattern of `new_data = self._mgr.<something>; self._constructor(new_data)`. In those cases, I think in theory the manager method should already have taken care of the references, and so an additional shallow copy is not needed. 

But, it should also be harmless (except for a bit of overheead), since those intermediate manager / blocks will go out of scope?

Reply:
I wanted to do this also for a manager, but  this caused all sorts of problems because we kept the Manager alive. 

Yeah this is a safety net for something like

```
new_data = self._mgr

if something:  # False
    ...
return self._constructor(new_data)
```

Reply:
Yes if they are a true intermediate manager they go out of scope immediately. But if we forgot performing a shallow copy for some reason, this catches this

Reply:
One more thing here, do you know what is the overhead of this shallow copy? Because our "fastpath" for DataFrame creation goes through here

Reply:
Small test (not using this branch):

```
In [97]: df = pd.DataFrame({'a': [1, 2, 3], 'b': [.1, .2, .3]})

In [98]: mgr = df._mgr

In [100]: %timeit mgr.copy(deep=False)
4.1 µs ± 47.4 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [102]: %timeit pd.DataFrame(mgr)
1.47 µs ± 35.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

So in _relative_ terms, it's not an insignificant change .. Not fully sure how important this is in real-world cases though.

Reply:
Yeah it's not totally cheap, it's from 1 to 3 on my machine. 

We could keep it out of here if you prefer and investigate other methods of keeping the reference here?